js/src/framework.ts

```diff
@@ -75,26 +97,36 @@ export async function Eval<Input, Output>(
   name: string,
   evaluator: Evaluator<Input, Output>
 ): Promise<void | ExperimentSummary> {
-  if (_evals[name]) {
-    throw new Error(`Evaluator ${name} already exists`);
+  const evalName = makeEvalName(name, evaluator.metadata);
+  if (_evals[evalName]) {
+    throw new Error(`Evaluator ${evalName} already exists`);
   }
   if (globalThis._lazy_load) {
-    _evals[name] = { name, ...evaluator };
+    _evals[evalName] = { evalName, projectName: name, ...evaluator };
```

Comment:
should we throw an error here if the name already exists? (I understand this is an existing bug/wart), not something new in your change

Reply:
Isn't this checked right above?

Reply:
oops 🤦